From: Hannes Reinecke <hare@suse.de>
Date: Mon, 24 Jun 2013 09:26:43 +0200
Subject: scsi: Fixup compilation warning
References: bnc#798050
Patch-Mainline: n/a

The compiler complained about 'variables not useable from userspace'.
Although he is right James B. deemed this message irrelevant.
But we should silence this message nevertheless.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 include/scsi/scsi.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/scsi/scsi.h b/include/scsi/scsi.h
index 8d696e2..bdfc15c 100644
--- a/include/scsi/scsi.h
+++ b/include/scsi/scsi.h
@@ -159,6 +159,7 @@ scsi_varlen_cdb_length(const void *hdr)
 	return ((struct scsi_varlen_cdb_hdr *)hdr)->additional_cdb_length + 8;
 }
 
+#ifdef __KERNEL__
 extern const unsigned char scsi_command_size_tbl[8];
 #define COMMAND_SIZE(opcode) scsi_command_size_tbl[((opcode) >> 5) & 7]
 
@@ -168,6 +169,7 @@ scsi_command_size(const unsigned char *cmnd)
 	return (cmnd[0] == VARIABLE_LENGTH_CMD) ?
 		scsi_varlen_cdb_length(cmnd) : COMMAND_SIZE(cmnd[0]);
 }
+#endif
 
 /*
  *  SCSI Architecture Model (SAM) Status codes. Taken from SAM-3 draft
@@ -284,8 +286,10 @@ enum scsi_protocol {
 	SCSI_PROTOCOL_UNSPEC = 0xf, /* No specific protocol */
 };
 
+#ifdef __KERNEL__
 /* Returns a human-readable name for the device */
 extern const char * scsi_device_type(unsigned type);
+#endif
 
 /*
  * standard mode-select header prepended to all mode-select commands
-- 
1.7.12.4

